chain/network/tests/routing.rs

```diff
@@ -571,3 +571,30 @@ fn churn_attack() {
     })
     .unwrap();
 }
+
+/// Spawn 4 nodes with max peers requird equal 2. Connect first three peers in a triangle.
+/// Try to connect peer3 and see it fail since
+#[test]
+fn max_peer_limit() {
+    init_test_logger();
+
+    System::run(|| {
+        let mut runner = Runner::new(4, 4, 2);
+
+        runner.push(Action::AddEdge(0, 1));
+        runner.push(Action::AddEdge(1, 2));
+        runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![2])]));
+        runner.push(Action::CheckRoutingTable(1, vec![(0, vec![0]), (2, vec![2])]));
+        runner.push(Action::CheckRoutingTable(2, vec![(1, vec![1]), (0, vec![0])]));
+        runner.push(Action::AddEdge(3, 0));
+        // TODO(Marx): After #1886 is merged use Action::Wait here
+        // runner.push(Action::Wait(10));
+        runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![2])]));
+        runner.push(Action::CheckRoutingTable(1, vec![(0, vec![0]), (2, vec![2])]));
+        runner.push(Action::CheckRoutingTable(2, vec![(1, vec![1]), (0, vec![0])]));
+        runner.push(Action::CheckRoutingTable(3, vec![]));
```

Comment:
where is the number of peers checked?

Reply:
There routing table for peer 3 is empty since connection failed to establish. 
`runner.push(Action::CheckRoutingTable(3, vec![]));` is enough to notice that peer3 have no route to any other peer in the system.